functions/utils/graphql-client.ts

```diff
@@ -0,0 +1,59 @@
+const MYCELIUM_API_URL = "https://mycelium.truffle.vip/graphql";
+
+interface MyceliumOptions {
+  accessToken: string;
+  orgId: string;
+  apiUrl: string;
+}
+
+export interface GQLResponse<T = unknown> {
+  data: T;
+  errors: unknown[];
+}
+
+export async function graphqlReq<T = unknown>(
+  query: string,
+  variables: Record<string, unknown>,
+  options: MyceliumOptions,
+) {
+  const result = await fetch(options.apiUrl, {
+    method: "POST",
+    body: JSON.stringify({ query, variables }),
+    headers: {
+      "Content-Type": "application/json",
+      "x-org-id": options.orgId,
+      "x-access-token": options.accessToken,
+    },
+  });
+
+  const resp = await result.json() as GQLResponse<T>;
+
+  if (resp?.errors?.length > 0) {
+    throw resp.errors;
+  }
+
+  return resp?.data;
+}
+
+export function gql(strings: TemplateStringsArray, ..._values: string[]) {
+  return strings.join();
+}
+
+export class MyceliumClient {
+  constructor(
+    public accessToken: string,
+    public orgId: string,
+    public apiUrl = MYCELIUM_API_URL,
+  ) {}
+
+  query = <T = unknown>(
```

Comment:
do we want to call this something different since this client could also be used to make mutations?

Reply:
we could also consider using a lib like: https://deno.land/x/graphql_request@v4.1.0